File: SharpCode/H00/S0001.cs
namespace SharpCode;

public class S0001 {
    public int[] TwoSum(int[] nums, int target) {
        var expect = new Dictionary<int, int>();
        for (int i = 0; i < nums.Length; i++) {
            if (expect.TryGetValue(nums[i], out int last)) {
                return new int[2] { last, i };
            }
            expect[target - nums[i]] = i;
        }
        return new int[2];
    }
}

File: .archived/ktcode/src/main/kotlin/h00/S0001.kt
package h00

class S0001 {
    fun twoSum(nums: IntArray, target: Int): IntArray {
        val map = mutableMapOf<Int, Int>()
        for ((i, v) in nums.withIndex()) {
            map[v]?.let { return intArrayOf(it, i)}
            map[target-v] = i
        }
        return intArrayOf()
    }
}